awx/main/models/unified_jobs.py

```diff
@@ -810,11 +817,18 @@ def save(self, *args, **kwargs):
 
         # Sanity check: Has the job just completed? If so, mark down its
         # completion time, and record its output to the database.
+        #import sdb
+        #sdb.set_trace()
         if self.status in ('successful', 'failed', 'error', 'canceled') and not self.finished:
-            # Record the `finished` time.
+            # Record the `finished` time and if the job was canceled, set it to null.
             self.finished = now()
+            if self.status == 'canceled':
+                self.finished = None
+                self.canceled = now()
             if 'finished' not in update_fields:
                 update_fields.append('finished')
+            if 'canceled' not in update_fields:
```

Comment:
You need to gate this conditional with and additional condition like `and not self.finished`, like we do for the finished field. It probably makes sense to move some of that logic for the finished event too.

Reply:
What would be an example that you would want to see @AlanCoding? Is the gate on line 822 not sufficient?

Reply:
If you only want this to run when the `finished` time is set, then I don't really follow what your objective is. That would make it a purely aesthetic change. The client can deduce all of this information now, with the knowledge that `finished` is roughly the time that the subprocess stops, whether it was canceled or not.